addons/arsenal/functions/fnc_removeBox.sqf

```diff
@@ -41,8 +41,8 @@ if (_global && {isMultiplayer} && {!isNil "_id"}) then {
     [QGVAR(boxRemoved), _object] call CBA_fnc_localEvent;
 };
 
-// If the arsenal is already open, close arsenal display
-if (!isNil QGVAR(currentBox) && {GVAR(currentBox) isEqualTo _object}) then {
+// If the arsenal is already open and not ignoring content (see FUNC(openBox)), close arsenal display
+if (!isNil QGVAR(currentBox) && {GVAR(currentBox) isEqualTo _object} && {isNil QGVAR(ignoredVirtualItems)}) then {
```

Comment:
Surely in this case the lazy eval with isNil doesn't save anything ?
```suggestion
if (!isNil QGVAR(currentBox) && {GVAR(currentBox) isEqualTo _object} && isNil QGVAR(ignoredVirtualItems)) then {
```

Reply:
It does.

Reply:
Care to elaborate ? I'd like to learn more.

Reply:
Test it out yourself, far better and more accurate than my explanations.